cmd/app/command.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	migrate "github.com/golang-migrate/migrate/v4"
+	"github.com/golang-migrate/migrate/v4/database/sqlite3"
+	_ "github.com/golang-migrate/migrate/v4/database/sqlite3"
+	_ "github.com/golang-migrate/migrate/v4/source/file"
+	_ "github.com/mattn/go-sqlite3"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/carfloresf/the-sun-god/config"
+	"github.com/carfloresf/the-sun-god/internal/router"
+	"github.com/carfloresf/the-sun-god/pkg/storage"
+	"github.com/carfloresf/the-sun-god/pkg/subreddit"
+)
+
+func execute(path string) {
+	config, err := config.NewConfig(path)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	storageDB, err := storage.NewStorage(&config.DB)
+	if err != nil {
+		log.Fatal("error creating storage: ", err)
+	}
+
+	driver, err := sqlite3.WithInstance(storageDB.Conn, &sqlite3.Config{})
+	if err != nil {
+		log.Fatal("error creating driver: %w", err)
+	}
+
+	m, err := migrate.NewWithDatabaseInstance(
+		"file://db/migrations",
+		"main", driver)
+	if err != nil {
+		log.Fatalf("migration error: %s", err)
+	}
+
+	if err = m.Up(); err != nil && err != migrate.ErrNoChange {
+		log.Errorf("migration error: %s", err)
+	}
+
+	err = storageDB.PrepareAllStatements()
+	if err != nil {
+		log.Fatalf("error preparing statements: %s", err)
+	}
+
+	router, err := router.InitRouter(storageDB)
+	if err != nil {
+		log.Fatal("router error", err)
+	}
+
+	server := &http.Server{
+		Addr:              ":" + config.HTTP.Port,
+		Handler:           router,
+		ReadTimeout:       5 * time.Second,
+		ReadHeaderTimeout: 5 * time.Second,
+	}
+
+	err = subreddit.LoadSubreddits(config.SRFile)
+	if err != nil {
+		log.Fatal("error loading subreddits: %w", err)
+	}
+
+	go func() {
+		if err := server.ListenAndServe(); err != nil && err != http.ErrServerClosed {
+			log.Fatalf("listen: %s\n", err)
+		}
+	}()
+
+	quit := make(chan os.Signal, 1)
+
+	signal.Notify(quit, syscall.SIGINT, syscall.SIGTERM)
+	<-quit
+	log.Println("shutting down server ...")
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
Why do you use context with timeout here? If we want graceful shutdown, why is it an issue if `server.Shutdown` takes more than 5s?

Reply:
In this case it is not an issue, but when we have more dependencies, sometimes we want to wait until everything closes.

Reply:
Sure but how would you choose 5s interval. It seems arbitrary. Everything should close sequentially, or if you want to do it in parallel, you should wait for the signal, not some arbitrary timeout. In your usecase, if some resources are failing to close within 5s, we are still going to quit
Just make sure that everything is done, by not doing the timeout, but rather wait for the notification either from the channel, or as a blocking call like Shutdown. If you don't provide cancelation context to it, it will return as soon as it completes everything. So if closing all network connection takes more than 5s, you are killing it with the context, instead of waiting for it

Reply:
Yeah 5s was kinda arbitrary, and yeah I'm "forcing" it to quit in 5s even if the server is not done.